fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -40,40 +124,102 @@ class SignalSparsity:
         dst_top_k_element (int, optional):
             Number of top-k elements to retain for DST. Default: None
         dst_top_k_percent (float, optional):
-            Percent of top-k elements to retain for DST. Default: 0.1
+            Percent of top-k elements to retain for DST. Default: None
 
     Example:
         .. code-block:: python
 
             2d_sparser = SignalSparsity()
             sst, dst = 2d_sparser.get_sst_dst(linear.weight.data)
 
-            3d_sparser = SingalSparsity(algo=Algo.DCT, sst_top_k_dim=None, dst_top_k_dim=-1, dst_top_k_element=5, dst_top_k_percent=None)
+            3d_sparser = SingalSparsity(algo=Algo.DCT, sst_top_k_dim=None, dst_top_k_dim=-1, sst_top_k_percent=10, dst_top_k_element=100)
             conv.weight.data = 3d_sparser.get_sst_dst_weight(conv.weight.data)
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(
+        self,
+        algo: Algo = Algo.FFT,
+        sst_top_k_dim: Optional[int] = -1,
+        sst_top_k_element: Optional[int] = None,
+        sst_top_k_percent: Optional[float] = None,
+        dst_top_k_dim: Optional[int] = -1,
+        dst_top_k_element: Optional[int] = None,
+        dst_top_k_percent: Optional[float] = None,
+    ) -> None:
+
+        self._sst_top_k_dim = sst_top_k_dim
+        self._sst_top_k_element = sst_top_k_element
+        self._sst_top_k_percent = sst_top_k_percent
+        self._dst_top_k_dim = dst_top_k_dim
+        self._dst_top_k_element = dst_top_k_element
+        self._dst_top_k_percent = dst_top_k_percent
+
         self._validate_conf()
+        # TODO: [Min] Type checking for the following
+        self._transform = torch.fft.fft if algo is Algo.FFT else _dct_transform  # type: ignore
 
     def _validate_conf(self) -> None:
-        """Validating the config is valid.
+        """Validating if the config is valid.
 
-        For example, not both top_k_element and top_k_percent is set.
+        This includes asserting the following:
+        1. validating that one and only one of top_k_element and top_k_percent is set.
+        2. Asserting that both element and percentage are in valid ranges.
 
-        this should assert fail if checking fails.
+        This asserts fail if the checking fails.
         """
-        pass
+        # assert that both top_k_elements and top_k_percent aren't set for sst and dst
+        def one_and_only(a: Optional[int], b: Optional[float]) -> bool:
+            return (a is None) ^ (b is None)
+
+        if not (
+            one_and_only(self._sst_top_k_element, self._sst_top_k_percent)
+            and one_and_only(self._dst_top_k_element, self._dst_top_k_percent)
+        ):
+            raise ValueError(
```

Comment:
put the conf var values in the error msg too, like:

raise ValueError(f".... Input values are: {self._sst_top_k_element} {...} {...} {...}")

Reply:
Was wondering if we also want to map the provided inputs to the arguments as well? e.g.
```
raise ValueError(f".... Input values are: sst_top_k_element = {self._sst_top_k_element} , ....<arg name> = {...}  ..")
```
Or just inputs are fine?

Reply:
sure, with names is better. using shorter names is also fine if the msg getting too long. It is mainly for easy diagnosis in case we receives an error report with this exception in the trace.

Reply:
Sounds good. Thanks!